source/core_mqtt.c

```diff
@@ -3747,16 +3747,16 @@ size_t MQTT_GetBytesInMQTTVec( MQTTVec_t * pVec )
 /*-----------------------------------------------------------*/
 
 void MQTT_SerializeMQTTVec( uint8_t * pAllocatedMem,
-                            MQTTVec_t * pVec )
+                            const MQTTVec_t * pVec )
 {
-    TransportOutVector_t * pTransportVec = pVec->pVector;
+    const TransportOutVector_t * pTransportVec = pVec->pVector;
     const size_t vecLen = pVec->vectorLen;
     size_t index = 0;
     size_t i = 0;
 
     for( i = 0; i < vecLen; i++ )
     {
-        memcpy( &pAllocatedMem[ index ], pTransportVec[ i ].iov_base, pTransportVec[ i ].iov_len );
+        ( void ) memcpy( &pAllocatedMem[ index ], ( const uint8_t * ) pTransportVec[ i ].iov_base, pTransportVec[ i ].iov_len );
```

Comment:
NIT: Since memcpy's signature is: `void* memcpy( void *dest, const void *src, size_t count );`
wouldn't it be better to cast it to `void`/`const void` pointers.
For example, we do it like this in +TCP: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/blob/main/source/FreeRTOS_DNS_Parser.c#L698-L704